src/Shared/Recruit.Vacancies.Client/Application/CommandHandlers/UpdateLiveVacancyCommandHandler.cs

```diff
@@ -33,15 +35,36 @@ public async Task Handle(UpdateLiveVacancyCommand message, CancellationToken can
         {
             _logger.LogInformation("Updating live vacancy {vacancyId}.", message.Vacancy.Id);
 
+            Vacancy originalVacancy = await _repository.GetVacancyAsync(message.Vacancy.Id);
+
             message.Vacancy.LastUpdatedDate = _timeProvider.Now;
             message.Vacancy.LastUpdatedByUser = message.User;
 
             await _repository.UpdateAsync(message.Vacancy);
 
+            await PublishLiveVacancyClosingDateChangedEvent(originalVacancy, message.Vacancy.ClosingDate);
```

Comment:
The below `VacancyPublishedEvent` will rebuild the projection that FAA uses to index and display the vacancy.  If we call `PublishLiveVacancyClosingDateChangedEvent` first then this could be handled by FAA before the vacancy projection has been updated. I don't know if this would be an issue however.  

Should the `PublishLiveVacancyClosingDateChangedEvent` be called only once the projection has been updated? I don't know. Maybe one for @mgwilliam

Reply:
I think FAA needs to react to `PublishLiveVacancyClosingDateChangedEvent`, to update the date on the application, in which case I don't see an issue if the search is updated later (with in an hour). We have accepted that lag anyway.